src/router.rs

```diff
@@ -100,11 +100,16 @@ impl RouteInfo {
     /// would return `false`.
     pub fn has_route(&self, path: &str) -> bool {
         let path = if path.is_empty() { "/" } else { path };
+        let ref prefix = self.router.prefix;
 
         for pattern in &self.router.patterns {
             if pattern.is_match(path) {
                 return true;
             }
+
+            if pattern.is_match(&format!("{}{}", prefix, path)) {
```

Comment:
You match against path above, and then you match again... against prefix and path together...
Should it depend on whether there is prefix or not?